presto-spi/src/main/java/com/facebook/presto/spi/ConnectorMaterializedViewDefinition.java

```diff
@@ -0,0 +1,254 @@
+            Map<String, Map<SchemaTableName, String>> originalColumnMapping,
+            Optional<List<String>> validRefreshColumns)
+    {
+        this(
+                originalSql,
+                schema,
+                table,
+                baseTables,
+                owner,
+                convertFromMapToColumnMappings(requireNonNull(originalColumnMapping, "originalColumnMapping is null"), new SchemaTableName(schema, table)),
+                validRefreshColumns);
+    }
+
+    @JsonProperty
+    public String getOriginalSql()
+    {
+        return originalSql;
+    }
+
+    @JsonProperty
+    public String getSchema()
+    {
+        return schema;
+    }
+
+    @JsonProperty
+    public String getTable()
+    {
+        return table;
+    }
+
+    @JsonProperty
+    public List<SchemaTableName> getBaseTables()
+    {
+        return baseTables;
+    }
+
+    @JsonProperty
+    public Optional<String> getOwner()
+    {
+        return owner;
+    }
+
+    @JsonProperty
+    public List<ColumnMapping> getColumnMappings()
+    {
+        return columnMappings;
+    }
+
+    @JsonProperty
+    public Optional<List<String>> getValidRefreshColumns()
+    {
+        return validRefreshColumns;
+    }
+
+    @Override
+    public String toString()
+    {
+        StringBuilder sb = new StringBuilder("ConnectorMaterializedViewDefinition{");
+        sb.append("originalSql=").append(originalSql);
+        sb.append(",schema=").append(schema);
+        sb.append(",table=").append(table);
+        sb.append(",baseTables=").append(baseTables);
+        sb.append(",owner=").append(owner.orElse(null));
+        sb.append(",columnMappings=").append(columnMappings);
+        sb.append(",validRefreshColumns=").append(validRefreshColumns.orElse(null));
+        sb.append("}");
+        return sb.toString();
+    }
+
+    @JsonIgnore
+    public Map<String, Map<SchemaTableName, String>> getColumnMappingsAsMap()
+    {
+        return columnMappings.stream()
+                .collect(toMap(
+                        mapping -> mapping.getViewColumn().getColumnName(),
+                        mapping -> mapping.getBaseTableColumns().stream().collect(toMap(TableColumn::getTableName, TableColumn::getColumnName))));
+    }
+
+    @JsonIgnore
+    private static List<ColumnMapping> convertFromMapToColumnMappings(Map<String, Map<SchemaTableName, String>> originalColumnMappings, SchemaTableName sourceTable)
+    {
+        List<ColumnMapping> columnMappingList = new ArrayList<>();
+
+        for (String sourceColumn : originalColumnMappings.keySet()) {
+            TableColumn viewColumn = new TableColumn(sourceTable, sourceColumn);
+
+            List<TableColumn> baseTableColumns = new ArrayList<>();
+            for (SchemaTableName baseTable : originalColumnMappings.get(sourceColumn).keySet()) {
+                baseTableColumns.add(new TableColumn(baseTable, originalColumnMappings.get(sourceColumn).get(baseTable)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.